File: array/array_lc76_test.go
package array

import (
	"testing"
)

func Test_minWindow(t *testing.T) {
	type args struct {
		s string
		t string
	}
	tests := []struct {
		name string
		args args
		want string
	}{
		{
			name: "lc case1",
			args: args{
				s: "ADOBECODEBANC",
				t: "ABC",
			},
			want: "BANC",
		},
		{
			name: "lc case2",
			args: args{
				s: "a",
				t: "a",
			},
			want: "a",
		},

		{
			name: "lc case3",
			args: args{
				s: "a",
				t: "aa",
			},
			want: "",
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := minWindow(tt.args.s, tt.args.t); got != tt.want {
				t.Errorf("minWindow() = %v, want %v", got, tt.want)
			}
		})
	}
}

func Test_minWindowPV1(t *testing.T) {
	type args struct {
		s string
		t string
	}
	tests := []struct {
		name string
		args args
		want string
	}{
		{
			name: "lc case1",
			args: args{
				s: "ADOBECODEBANC",
				t: "ABC",
			},
			want: "BANC",
		},
		{
			name: "lc case2",
			args: args{
				s: "a",
				t: "a",
			},
			want: "a",
		},

		{
			name: "lc case3",
			args: args{
				s: "a",
				t: "aa",
			},
			want: "",
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := minWindowPV1(tt.args.s, tt.args.t); got != tt.want {
				t.Errorf("minWindowPV1() = %v, want %v", got, tt.want)
			}
		})
	}
}

File: tests/array_py/array_lc76_test.py
import pytest
from tests.helpers import Case
from array_py.array_lc76 import Solution

CASES_minWindow = [
    Case(name="basic", args=("ADOBECODEBANC", "ABC"), want="BANC"),
    Case(name="base 1", args=("a", "a"), want="a"),
    Case(name="base 1", args=("a", "aa"), want=""),
    Case(name="base 3", args=("ab", "b"), want="b"),
]

@pytest.mark.parametrize("case", CASES_minWindow, ids=lambda c: c.name)
def test_minWindow(case: Case):
    s = Solution()

    if case.raises:
        with pytest.raises(case.raises):
            s.minWindow(*case.args, **case.kwargs)
        return

    got = s.minWindow(*case.args, **case.kwargs)
    assert got == case.want

CASES_minWindowV2 = [
    Case(name="basic", args=("ADOBECODEBANC", "ABC"), want="BANC"),
    Case(name="base 1", args=("a", "a"), want="a"),
    Case(name="base 1", args=("a", "aa"), want=""),
    Case(name="base 3", args=("ab", "b"), want="b"),
]

@pytest.mark.parametrize("case", CASES_minWindowV2, ids=lambda c: c.name)
def test_minWindowV2(case: Case):
    s = Solution()

    if case.raises:
        with pytest.raises(case.raises):
            s.minWindowV2(*case.args, **case.kwargs)
        return

    got = s.minWindowV2(*case.args, **case.kwargs)
    assert got == case.want

